src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -447,17 +447,18 @@ grpc_error_handle XdsResolver::XdsConfigSelector::CreateMethodConfig(
   if (route_action.retry_policy.has_value() &&
       !route_action.retry_policy->retry_on.Empty()) {
     std::vector<std::string> retry_parts;
+    const auto base_interval =
+        route_action.retry_policy->retry_back_off.base_interval.as_timespec();
+    const auto max_interval =
+        route_action.retry_policy->retry_back_off.max_interval.as_timespec();
     retry_parts.push_back(absl::StrFormat(
         "\"retryPolicy\": {\n"
         "      \"maxAttempts\": %d,\n"
         "      \"initialBackoff\": \"%d.%09ds\",\n"
```

Comment:
There are a number of places where we format a duration this way, since it's [the JSON form of a `google.protobuf.Duration` proto](https://developers.google.com/protocol-buffers/docs/proto3#json).  It looks like we're also doing this in rls_end2end_test.cc, and probably some other places.  How about having a method on `Duration` that outputs this format, so that we don't need to handle it at each individual call site?

Reply:
Happy to add a string conversion method, but maybe in a secondary PR?

Reply:
Sure, separate PR is fine.